examples/native/native_test.c

```diff
@@ -1111,21 +1229,36 @@ int TPM2_Native_Test(void* userCtx)
     }
 
     /* Close object handle */
-    cmdIn.flushCtx.flushHandle = handle;
-    TPM2_FlushContext(&cmdIn.flushCtx);
-    cmdIn.flushCtx.flushHandle = eccKey.handle;
-    TPM2_FlushContext(&cmdIn.flushCtx);
-    cmdIn.flushCtx.flushHandle = hmacKey.handle;
-    TPM2_FlushContext(&cmdIn.flushCtx);
-    cmdIn.flushCtx.flushHandle = rsaKey.handle;
-    TPM2_FlushContext(&cmdIn.flushCtx);
+    if (handle != TPM_RH_NULL) {
+        cmdIn.flushCtx.flushHandle = handle;
+        TPM2_FlushContext(&cmdIn.flushCtx);
+    }
+    if (eccKey.handle != TPM_RH_NULL) {
+        cmdIn.flushCtx.flushHandle = eccKey.handle;
+        TPM2_FlushContext(&cmdIn.flushCtx);
+    }
+    if (hmacKey.handle != TPM_RH_NULL) {
+        cmdIn.flushCtx.flushHandle = hmacKey.handle;
+        TPM2_FlushContext(&cmdIn.flushCtx);
+    }
+    if (aesKey.handle != TPM_RH_NULL) {
+        cmdIn.flushCtx.flushHandle = aesKey.handle;
+        TPM2_FlushContext(&cmdIn.flushCtx);
+    }
+    if (rsaKey.handle != TPM_RH_NULL) {
+        cmdIn.flushCtx.flushHandle = rsaKey.handle;
+        TPM2_FlushContext(&cmdIn.flushCtx);
+    }
 
     /* Cleanup key handles */
-    cmdIn.flushCtx.flushHandle = endorse.handle;
-    TPM2_FlushContext(&cmdIn.flushCtx);
-    cmdIn.flushCtx.flushHandle = storage.handle;
-    TPM2_FlushContext(&cmdIn.flushCtx);
-
+    if (endorse.handle != TPM_RH_NULL) {
+        cmdIn.flushCtx.flushHandle = endorse.handle;
+        TPM2_FlushContext(&cmdIn.flushCtx);
+    }
+    if (storage.handle != TPM_RH_NULL) {
+        cmdIn.flushCtx.flushHandle = storage.handle;
+        TPM2_FlushContext(&cmdIn.flushCtx);
```

Comment:
do we need a check here on the return value and on other calls to TPM2_FlushContext